### PR TITLE
cloud integration: add dbt doc in integration section

### PR DESCRIPTION
**Pull request by Daemonxiao**
### First-time contributors' checklist 

- [x] I've signed [**Contributor License Agreement**](https://cla-assistant.io/pingcap/docs) that's required for repo owners to accept my contribution.

### What is changed, added or deleted? (Required)

Add dbt doc.

### Which TiDB version(s) do your changes apply to? (Required)

**Tips for choosing the affected version(s):**

By default, **CHOOSE MASTER ONLY** so your changes will be applied to the next TiDB major or minor releases. If your PR involves a product feature behavior change or a compatibility change, **CHOOSE THE AFFECTED RELEASE BRANCH(ES) AND MASTER**.

For details, see [tips for choosing the affected versions](https://github.com/pingcap/docs/blob/master/CONTRIBUTING.md#guideline-for-choosing-the-affected-versions).

- [ ] master (the latest development version)
- [ ] v6.4 (TiDB 6.4 versions)
- [ ] v6.3 (TiDB 6.3 versions)
- [x] v6.1 (TiDB 6.1 versions)
- [ ] v5.4 (TiDB 5.4 versions)
- [ ] v5.3 (TiDB 5.3 versions)
- [ ] v5.2 (TiDB 5.2 versions)
- [ ] v5.1 (TiDB 5.1 versions)
- [ ] v5.0 (TiDB 5.0 versions)

### What is the related PR or file link(s)?

- This PR is translated from:
- Other reference link(s):

### Do your changes match any of the following descriptions?

- [ ] Delete files
- [ ] Change aliases
- [ ] Need modification after applied to another branch 
- [ ] Might cause conflicts after applied to another branch

### CODE REVIEW

File: tidb-cloud/dbt.md [hunk 1]
```diff
@@ -0,0 +1,223 @@
+---
+title: TiDB with dbt
+summary: Learn the use cases of dbt in TiDB.
+---
+
+# TiDB with dbt
+
+[Data build tool (dbt)](https://www.getdbt.com/) is a popular open-source data transformation tool that enables analytics engineers to transform data in their warehouses through SQL statements. Through the [dbt-tidb](https://github.com/pingcap/dbt-tidb) plug-in, analytics engineers working with TiDB can directly create forms and match data through SQL without having to think about the process of creating tables or views.
```
Comment (shiyuhang0) [inline tidb-cloud/dbt.md:8]: Ask reader to read dbt offical doc first. I think a user can't understand this doc before he read the offical doc

File: tidb-cloud/dbt.md [hunk 2]
```diff
@@ -0,0 +1,223 @@
...
+
+## Installation
+
+Installing dbt and dbt-tidb requires only one command because dbt is installed as a dependency when we install dbt-tidb.
+
+```shell
+pip install dbt-tidb
+```
+
+## Creating project: jaffle shop
```
Comment (shiyuhang0) [inline tidb-cloud/dbt.md:25]: 只是一个想法：咱们的官方文档，要用 jaffle shop 这个例子吗？还是从头开始一个例子？

File: tidb-cloud/dbt.md [hunk 3]
```diff
@@ -0,0 +1,229 @@
+---
+title: TiDB with dbt
+summary: Learn the use cases of dbt in TiDB.
+---
+
+# TiDB with dbt
```
Comment (hfxsd) [inline tidb-cloud/dbt.md:6]: Use dbt-tidb to Create Tables and Views for TiDB

File: tidb-cloud/dbt.md [hunk 4]
```diff
@@ -0,0 +1,229 @@
...
+[Data build tool (dbt)](https://www.getdbt.com/) is a popular open-source data transformation tool that enables analytics engineers to transform data in their warehouses through SQL statements. Through the [dbt-tidb](https://github.com/pingcap/dbt-tidb) plug-in, analytics engineers working with TiDB can directly create forms and match data through SQL without having to think about the process of creating tables or views.
+
+This document uses the official dbt tutorial as an example to introduce how to use TiDB in dbt together.
+
+The software used in this example and its version requirements:
+
+- TiDB 5.3 or later versions
+- dbt 1.01 or later versions
+- dbt-tidb 1.0.0
+
```
Comment (hfxsd) [inline tidb-cloud/dbt.md:17]: Make the following sections steps?

File: tidb-cloud/dbt.md [hunk 5]
```diff
@@ -0,0 +1,229 @@
...
+## Generate the docs
+
+Use the following command to generate a visual document:
+
+```
+dbt docs generate
+dbt docs serve
+```
+
+The document contains the overall structure of the jaffle_shop project and a description of all the tables and views. You can viewed it at [http://localhost:8080](http://localhost:8080).
```
Comment (hfxsd) [inline tidb-cloud/dbt.md:179]:
```suggestion
The document contains the overall structure of the jaffle_shop project and descriptions of all the tables and views. You can view it at [http://localhost:8080](http://localhost:8080).
```

File: tidb-cloud/dbt.md [hunk 6]
```diff
@@ -0,0 +1,229 @@
...
+- cast_bool_to_text
+- concat
+- escape_single_quotes
+- except
+- intersect
+- length
+- position
+- replace
+- right
+- listagg (not support yet)
```
Comment (hfxsd) [inline tidb-cloud/dbt.md:225]:
```suggestion
- listagg (not supported yet)
```

File: tidb-cloud/dbt.md [hunk 7]
```diff
@@ -0,0 +1,229 @@
...
+
+This document uses the official dbt tutorial as an example to introduce how to use TiDB in dbt together.
+
+The software used in this example and its version requirements:
+
+- TiDB 5.3 or later versions
+- dbt 1.01 or later versions
+- dbt-tidb 1.0.0
+
+## Install dbt
```
Comment (hfxsd) [inline tidb-cloud/dbt.md:18]:
```suggestion
## Step 1: Install dbt
```

File: tidb-cloud/dbt.md (hunk 1 above)
Comment (Daemonxiao) [reply tidb-cloud/dbt.md:8]: This is a tutorial about dbt-tidb. We have already given the dbt website here, so we need not advice readers to read dbt official doc first.

File: tidb-cloud/dbt.md (hunk 2 above)
Comment (Daemonxiao) [reply tidb-cloud/dbt.md]: 从头开始还要准备相关数据、导入数据，反而加重用户尝试的负担。

**Review by shiyuhang0 — APPROVED** [review]

**Comment by shiyuhang0** [comment]
please fix the failed ci

File: tidb-cloud/integrate-tidbcloud-with-dbt.md [hunk 8]
```diff
@@ -0,0 +1,317 @@
...
+    jaffle_shop_tidb:                                                         # project name
+      target: dev                                                             # target
+      outputs:
+        dev:
+          type: tidb                                                          # adapter type
+          server: gateway01.ap-southeast-1.prod.aws.tidbcloud.com             # endpoint
+          port: 4000
+          schema: analytics                                                   # database name
+          username: xxxxxxxxxxx.root
+          password: "your_password"
```
Comment (ran-huang) [inline tidb-cloud/integrate-tidbcloud-with-dbt.md:84]: 这里应该提示 要根据 TiDB cloud 的 setup 来配置，或者链接到后面的“## Description of profile fields”

File: tidb-cloud/integrate-tidbcloud-with-dbt.md [hunk 9]
```diff
@@ -0,0 +1,317 @@
...
+    mysql> show tables;
+    +---------------------+
+    | Tables_in_analytics |
+    +---------------------+
+    | raw_customers       |
+    | raw_orders          |
+    | raw_payments        |
+    +---------------------+
+    3 rows in set (0.00 sec)
+    ```
```
Comment (ran-huang) [inline tidb-cloud/integrate-tidbcloud-with-dbt.md:182]: 这里可以加一句 `select * from raw_customers limit 10;` 吗？让用户看到跟后面转换后的表数据的对比

File: tidb-cloud/integrate-tidbcloud-with-dbt.md [hunk 10]
```diff
@@ -0,0 +1,317 @@
...
+- The **seeds** directory stores the CSV files that are dumped from the database export tools. For example, TiDB Cloud can export the table data into CSV files through [Dumpling](https://docs.pingcap.com/tidbcloud/dumpling-overview#dumpling-overview). In the jaffle_shop project, these CSV files are used as raw data to be processed.
+
+## Step 3: Configure the project
+
+To configure the project:
+
+1. Complete the global configuration. In the user directory, edit the default global profile, `~/.dbt/profiles.yml` to configure the connection with TiDB Cloud:
+
+    ```shell
+    vi ~/.dbt/profiles.yml
```
[the remark this answers is not in the document]
Comment (Daemonxiao) [reply tidb-cloud/integrate-tidbcloud-with-dbt.md:74]: 试一下 sudo。
`sudo vi ~/.dbt/profiles.yml`

File: tidb-cloud/integrate-tidbcloud-with-dbt.md [hunk 11]
```diff
@@ -0,0 +1,317 @@
...
+This document uses a project in the official dbt tutorial as an example to introduce how to use dbt with TiDB Cloud.
+
+## Step 1: Install dbt and dbt-tidb
+
+Installing dbt and dbt-tidb requires only one command because dbt is installed as a dependency when you install dbt-tidb.
+
+```shell
+pip install dbt-tidb
+```
+
```
[the remark this answers is not in the document]
Comment (Daemonxiao) [reply tidb-cloud/integrate-tidbcloud-with-dbt.md:19]: 试了下，dbt 更新版本了，现在使用 --version 确实会显示不兼容，但是可以使用。后面应该会做 dbt-core 1.3 兼容。

File: tidb-cloud/integrate-tidbcloud-with-dbt.md [hunk 12]
```diff
@@ -0,0 +1,316 @@
...
+    mysql> show tables;
+    +---------------------+
+    | Tables_in_analytics |
+    +---------------------+
+    | raw_customers       |
+    | raw_orders          |
+    | raw_payments        |
+    +---------------------+
+    3 rows in set (0.00 sec)
+    ```
```
Comment (Daemonxiao) [inline tidb-cloud/integrate-tidbcloud-with-dbt.md:181]:
```suggestion

    mysql> select * from raw_customers limit 10;
    +------+------------+-----------+
    | id   | first_name | last_name |
    +------+------------+-----------+
    |    1 | Michael    | P.        |
    |    2 | Shawn      | M.        |
    |    3 | Kathleen   | P.        |
    |    4 | Jimmy      | C.        |
    |    5 | Katherine  | R.        |
    |    6 | Sarah      | R.        |
    |    7 | Martin     | M.        |
    |    8 | Frank      | R.        |
    |    9 | Jennifer   | F.        |
    |   10 | Henry      | W.        |
    +------+------------+-----------+
    10 rows in set (0.10 sec)
    ```
```

File: tidb-cloud/integrate-tidbcloud-with-dbt.md (hunk 9 above)
Comment (Daemonxiao) [reply tidb-cloud/integrate-tidbcloud-with-dbt.md:198]: done

File: tidb-cloud/integrate-tidbcloud-with-dbt.md [hunk 13]
```diff
@@ -0,0 +1,333 @@
...
+- The `seeds` directory stores the CSV files that are dumped by the database export tools. For example,  you can export the TiDB Cloud data into CSV files through [Dumpling](https://docs.pingcap.com/tidbcloud/dumpling-overview#dumpling-overview). In the `jaffle_shop` project, these CSV files are used as raw data to be processed.
+
+## Step 3: Configure the project
+
+To configure the project:
+
+1. Complete the global configuration. In the user directory, edit the default global profile, `~/.dbt/profiles.yml` to configure the connection with TiDB Cloud:
+
+    ```shell
+    vi ~/.dbt/profiles.yml
```
Comment (Daemonxiao) [inline tidb-cloud/integrate-tidbcloud-with-dbt.md:73]:
```suggestion
    sudo vi ~/.dbt/profiles.yml
```

File: tidb-cloud/integrate-tidbcloud-with-dbt.md (hunk 8 above)
Comment (Daemonxiao) [reply tidb-cloud/integrate-tidbcloud-with-dbt.md:83]: 已添加`Description of profile fields`链接

**Review by hfxsd — APPROVED** [review]

File: tidb-cloud/integrate-tidbcloud-with-dbt.md [hunk 14]
```diff
@@ -0,0 +1,333 @@
...
+    jaffle_shop_tidb:                                                         # project name
+      target: dev                                                             # target
+      outputs:
+        dev:
+          type: tidb                                                          # adapter type
+          server: gateway01.ap-southeast-1.prod.aws.tidbcloud.com             # endpoint
+          port: 4000
+          schema: analytics                                                   # database name
+          username: xxxxxxxxxxx.root
+          password: "your_password"
```
[the remark this answers is not in the document]
Comment (hfxsd) [reply tidb-cloud/integrate-tidbcloud-with-dbt.md]: Done

File: tidb-cloud/integrate-tidbcloud-with-dbt.md [hunk 15]
```diff
@@ -0,0 +1,333 @@
...
+            materialized: view           # *.sql which in models/staging/ would bt materialized to view
+    ```
+
+3. Verify the configuration. Run the following command to check whether the database and project configuration is correct.
+
+    ```shell
+    dbt debug
+    ```
+
+## Step 4: Load CSV files
```
Comment (ran-huang) [inline tidb-cloud/integrate-tidbcloud-with-dbt.md:125]:
```suggestion
## Step 4: (optional) Load CSV files
```

File: tidb-cloud/integrate-tidbcloud-with-dbt.md [hunk 16]
```diff
@@ -0,0 +1,333 @@
...
+
+Now that you have successfully created and configured the project, it’s time to load the CSV data and materialize the CSV as a table in the target database. Note that this step is not generally required for a dbt project because the data for processing is already in the database.
```
Comment (ran-huang) [inline tidb-cloud/integrate-tidbcloud-with-dbt.md:127]:
```suggestion
> **Note:**
>
> This step is optional. If the data for processing is already in the target database, you can skip this step.

Now that you have successfully created and configured the project, it’s time to load the CSV data and materialize the CSV as a table in the target database. 
```

File: tidb-cloud/integrate-tidbcloud-with-dbt.md [hunk 17]
```diff
@@ -0,0 +1,333 @@
...
+    1 of 3 OK loaded seed file analytics.raw_customers.............................. [INSERT 100 in 0.19s]
+    2 of 3 START seed file analytics.raw_orders..................................... [RUN]
+    2 of 3 OK loaded seed file analytics.raw_orders................................. [INSERT 99 in 0.14s]
+    3 of 3 START seed file analytics.raw_payments................................... [RUN]
+    3 of 3 OK loaded seed file analytics.raw_payments............................... [INSERT 113 in 0.24s]
+    ```
+
+    As you can see in the results, the seed file was started and loaded into three tables: `analytics.raw_customers`, `analytics.raw_orders`, and `analytics.raw_payments`.
+
+2. Verify the results in TiDB Cloud. The `show databases` command lists the new analytics database that dbt created. The `show tables` command indicates that there are three tables in the analytics database, corresponding to the ones you have created.
```
Comment (ran-huang) [inline tidb-cloud/integrate-tidbcloud-with-dbt.md:154]:
```suggestion
2. Verify the results in TiDB Cloud. The `show databases` command lists the new `analytics` database that dbt created. The `show tables` command indicates that there are three tables in the `analytics` database, corresponding to the ones you have created.
```

File: tidb-cloud/integrate-tidbcloud-with-dbt.md [hunk 18]
```diff
@@ -0,0 +1,333 @@
...
+    |    6 | Sarah      | R.        |
+    |    7 | Martin     | M.        |
+    |    8 | Frank      | R.        |
+    |    9 | Jennifer   | F.        |
+    |   10 | Henry      | W.        |
+    +------+------------+-----------+
+    10 rows in set (0.10 sec)
+    ```
+
+## Step 5: Run the project
```
Comment (ran-huang) [inline tidb-cloud/integrate-tidbcloud-with-dbt.md:200]:
```suggestion
## Step 5: Transform data
```

File: tidb-cloud/integrate-tidbcloud-with-dbt.md [hunk 19]
```diff
@@ -0,0 +1,333 @@
...
+    Finished running 3 view models, 2 table models in 2.27s.
+
+    Completed successfully
+
+    Done. PASS=5 WARN=0 ERROR=0 SKIP=0 TOTAL=5
+    ```
+
+    The result shows that two tables (`analytics.customers` and `analytics.orders`), and three views (`analytics.stg_customers`, `analytics.stg_orders`, and `analytics.stg_payments`) are created successfully.
+
+2. Go to TiDB Cloud to verify that the creation is successful.
```
Comment (ran-huang) [inline tidb-cloud/integrate-tidbcloud-with-dbt.md:238]:
```suggestion
2. Go to TiDB Cloud to verify that the transformation is successful.
```

File: tidb-cloud/integrate-tidbcloud-with-dbt.md [hunk 20]
```diff
@@ -0,0 +1,333 @@
...
+| server           | The TiDB Cloud clusters' endpoint to connect to                         | Required  | `gateway01.ap-southeast-1.prod.aws.tidbcloud.com` |
+| port             | The port to use                                                         | Required  | `4000`                                            |
+| schema           | Specify the schema (database) to normalize data into                      | Required  | `analytics`                                       |
+| username         | The username to use to connect to the TiDB Cloud clusters               | Required  | `xxxxxxxxxxx.root`                                |
+| password         | The password to use for authenticating to the TiDB Cloud clusters       | Required  | `"your_password"`                                 |
+| retries          | The retry times for connection to TiDB Cloud clusters (1 by default)    | Optional  | `2`                                               |
+
+## Supported functions
+
+The cross-db macros are moved from dbt-utils into dbt-core, so you can use the following functions directly. For information about how to use them, see [dbt-util](https://github.com/dbt-labs/dbt-utils).
```
Comment (ran-huang) [inline tidb-cloud/integrate-tidbcloud-with-dbt.md:311]: 这里感觉没有 cross-db 的上下文，用户不需要知道什么是 dbt-utils。能否直接写：

```suggestion
You can use the following functions directly in dbt-tidb. For information about how to use them, see [dbt-util](https://github.com/dbt-labs/dbt-utils).
```

Comment (Daemonxiao) [reply tidb-cloud/integrate-tidbcloud-with-dbt.md]: 好的，已修改。

File: tidb-cloud/integrate-tidbcloud-with-dbt.md [hunk 21]
```diff
@@ -0,0 +1,356 @@
...
+- `cast_bool_to_text`
+- `concat`
+- `escape_single_quotes`
+- `except`
+- `intersect`
+- `length`
+- `position`
+- `replace`
+- `right`
+- `listagg` (not supported yet)
```
Comment (ran-huang) [inline tidb-cloud/integrate-tidbcloud-with-dbt.md:356]: listagg 到底是支持了还是没支持，L336 和 L356 矛盾的 😂

Comment (Daemonxiao) [reply tidb-cloud/integrate-tidbcloud-with-dbt.md:356]: 了解了。我把这条去掉，本来这里只是列出所有方法的。

Comment (Daemonxiao) [inline tidb-cloud/integrate-tidbcloud-with-dbt.md:356]:
```suggestion
```

**Review by ran-huang — APPROVED** [review]
LGTM

**Pull request merged**